src/solver/simulation/economy.cpp

```diff
@@ -168,10 +169,13 @@ bool Economy::simulationBegin()
                 logs.fatal() << "internal error";
                 return false;
             }
-        }
 
-        weeklyOptProblem
-          = interfaceWeeklyOptimization::create(study.parameters.adqPatch.enabled, pProblemesHebdo);
+            weeklyOptProblems_.push_back(
+                    interfaceWeeklyOptimization::create(study.parameters.adqPatch.enabled,
+                                                        pProblemesHebdo[numSpace], 
+                                                        numSpace)
```

Comment:
I'm not sure quadratic opt are run in parallel, to be checked. It would be nice if they were.

With the current  implementation, there is no guarantee that `weeklyOptProblems_` contains the problem associated to index `numSpace`. I'd prefer something like this
```cpp
// before the for loop
weeklyOptProblems_.resize(pNbMaxPerformedYearsInParallel);
// inside the for loop
weeklyOptProblems_[numSpace] = interfaceWeeklyOptimization::create(/* ... */);
```

Reply:
ok, I'll check for the quadratic optimization.
For the rest of your remark, I was thinking that the creation of a **PROBLEME_HEBDO** could even be done at creation of a **WeeklyOptimization**.  i guess it would make sense. What do you think ?
We would have something like : 
```C++
bool Economy::simulationBegin()
{
    if (!preproOnly)
    {
        for (uint numSpace = 0; numSpace < pNbMaxPerformedYearsInParallel; numSpace++)
        {

            weeklyOptProblems_.push_back(
                    interfaceWeeklyOptimization::create(
                                study.parameters.adqPatch.enabled,
                                numSpace)
            );
        }
        SIM_InitialisationResultats();
    }

    pStartTime = study.calendar.days[study.parameters.simulationDays.first].hours.first;
    pNbWeeks = (study.parameters.simulationDays.end - study.parameters.simulationDays.first) / 7;
    return true;
}
```

Reply:
Yes, that's too ambitious for now. See also **adequacy.cpp**

Reply:
about `weeklyOptProblems_.resize(...)`, done